juju/juju
# [JUJU-1949] Dqlite DB worker connection caching

The proof-of-concept implementation of the `dbaccessor` worker had two methods, `GetDB` and `GetExistingDB`. `GetDB` was intended as a call through to Dqlite's `Open` method, with `GetExistingDB` used to acquire a connection to a previously opened DB.

As an exported API surface, this doesn't really make sense as it gives any consumer the ability to open (or more importantly create) any database.

Instead, we take the approach of assuming that the DB worker will know what databases it needs to open and cache connections for. `GetExistingDB` is removed in favour of `GetDB` only returning cached database connections, or an error if no such DB is known.

The first worker and manifold tests are introduced, with attendant indirections and mocks.

## QA steps

Acceptance tests will accompany the patch following this one, where `GetDB` is used to supply a DB to the new lease store. This patch is verified by unit tests, which can be run as follows.
- Run `make dqlite-deps-pull`.
- Create a file in the project root, _test.list_, with the following contents:
```
github.com/juju/juju/database
github.com/juju/juju/database/migration
github.com/juju/juju/database/repository
github.com/juju/juju/worker/dbaccessor
```
- Run `TEST_PACKAGE_LIST=test.list make test` and ensure tests pass.

## Documentation changes

None.

## Bug reference

N/A
